cms/djangoapps/contentstore/views/component.py

```diff
@@ -442,22 +441,29 @@ def component_handler(request, usage_key_string, handler, suffix=''):
             django response
     """
     usage_key = UsageKey.from_string(usage_key_string)
-    if is_xblock_aside(usage_key):
-        return invoke_xblock_aside_handler(request, usage_key, handler, suffix=suffix)
 
     # Let the module handle the AJAX
     req = django_to_webob_request(request)
 
     try:
-        descriptor = modulestore().get_item(usage_key)
-        descriptor.xmodule_runtime = StudioEditModuleRuntime(request.user)
-        resp = descriptor.handle(handler, req, suffix)
+        if is_xblock_aside(usage_key):
+            # Get the descriptor for the block being wrapped by the aside (not the aside itself)
+            descriptor = modulestore().get_item(usage_key.usage_key)
+            aside_instance = get_aside_from_xblock(descriptor, usage_key.aside_type)
+            aside_instance.runtime = StudioEditModuleRuntime(request.user)
+            asides = [aside_instance] if aside_instance else []
```

Comment:
I'm not sure if you need the if clause here, since if it was `None` the previous line would fail